js/scripts.js

```diff
@@ -1,94 +1,116 @@
+			pokemonList.push(pokemon);
 		} else {
 			return 'invalid input';
 		}
 	}
-	// function to show all movies
+	// function to show all pokemons
 	function getAll() {
-		return movieList;
+		return pokemonList;
 	}
 
-	// function to find a specific movie
-	function findMovie(userInput) {
-		return movieList.filter((movie) => movie.name.toLowerCase() === userInput.toLowerCase());
+	// function to find a specific pokemon
+	function findPokemon(userInput) {
+		return pokemonList.filter((pokemon) => pokemon.name.toLowerCase() === userInput.toLowerCase());
 	}
 
-	//function which creates buttons for each movie including showDetails of movie by click
-	function addListItem(movie) {
-		let list = document.querySelector('.movie-list');
+	//function which creates buttons for each pokemon including showDetails of pokemon by click
+	function addListItem(pokemon) {
+		let list = document.querySelector('.pokemon-list');
 		let listItem = document.createElement('li');
 		let button = document.createElement('button');
-		button.innerText = movie.name;
+		button.innerText = pokemon.name;
 		button.classList.add('name-button');
 		listItem.appendChild(button);
 		list.appendChild(listItem);
 		list.classList.add('list-class');
 		button.classList.add('button-class');
-		button.addEventListener('click', function() {
-			showDetails(movie);
+		button.addEventListener('click', function(event) {
+			showDetails(pokemon);
 		});
 	}
 
-	function showDetails(movie) {
-		console.log(movie);
+	function loadList() {
+		loadingMessage();
+		return fetch(apiUrl)
+			.then(function(response) {
+				hidingMessage();
+				return response.json();
+			})
+			.then(function(json) {
+				json.results.forEach(function(item) {
+					let pokemon = {
+						name: item.name,
+						detailsUrl: item.url
+					};
+					add(pokemon);
+					console.log(pokemon);
+				});
+			})
+			.catch(function(e) {
+				hidingMessage();
+				console.error(e);
+			});
+	}
+
+	function loadDetails(item) {
+		loadingMessage();
+		let url = item.detailsUrl;
+		return fetch(url)
+			.then(function(response) {
+				hidingMessage();
+				return response.json();
+			})
+			.then(function(details) {
+				item.imgUrl = details.sprites.front_default;
+				item.height = details.height;
+				item.types = details.types;
+			})
+			.catch(function(e) {
+				hidingMessage();
+				console.error(e);
+			});
 	}
 
-	// image function
-	function images(image) {
-		let img = document.createElement('img');
-		img.src = image;
-		let listItem = document.querySelector('.list-class');
-		listItem.appendChild(img);
+	function showDetails(item) {
+		a24Repository.loadDetails(item).then(function() {
+			console.log(item);
+		});
 	}
 
 	return {
 		add: add,
 		getAll: getAll,
-		findMovie: findMovie,
 		addListItem: addListItem,
+		loadList: loadList,
+		loadDetails: loadDetails,
 		showDetails: showDetails,
-		images: images
+		loadingMessage: loadingMessage,
+		hidingMessage: hidingMessage
```

Comment:
Here we should only expose functions that we need to use otherwise they should remain private. So `loadList, loadDetails, showDetails, loadingMessage, hidingMessage` should all remain private and not added to the return.